gc/base/VirtualMemory.cpp

```diff
@@ -125,6 +125,11 @@ MM_VirtualMemory::initialize(MM_EnvironmentBase* env, uintptr_t size, void* pref
 			freeMemory();
 			_heapBase = NULL;
 		}
+#if defined(OMR_GC_DOUBLE_MAP_ARRAYLETS)
+		if (0 != (OMRPORT_VMEM_MEMORY_MODE_DOUBLE_MAP_AVAILABLE & _identifier.mode)) {
+			_extensions->isDoubleMapAPIAvailable = true;
+		}
+#endif /* defined(OMR_GC_DOUBLE_MAP_ARRAYLETS) */
```

Comment:
`MM_VirtualMemory` instances are created not only for Object Heap but for many GC Metadata structures (Mark Map, Card Table etc). If this setting relates to Object Heap only there is a wrong place to call

Reply:
Yes, you're correct. The trick here is that, based on a decision made in the port library, weather the heap is created with `memfd_create` or not, needs to be propagated back so that we can enable or disable double mapping. My thought here was to steal a bit from the `mode` field for the heap `_identifier` and store it there with the flag `OMRPORT_VMEM_MEMORY_MODE_DOUBLE_MAP_AVAILABLE`. And the only place where both `_extensions` and heap `_identifier` is available is `MM_VirtualMemory` where I set `isDoubleMapAPIAvailable` to true. Another way would be to add a field to `OMR_VM`,  say `BOOLEAN isDoubleMapAPIAvailable`, and set that accordingly. There might a be another way that I might be missing?

Reply:
The `MM_VirtualMemory` instance is stored with correspondent structure, including `_identifier`. You just need to add new public function to `MM_VirtualMemory` to do this check and call it where you need to know an answer (or at least if Object Heap specific code)

Reply:
Yeah that'll also work, thanks!